### PR TITLE
ISPN-12824 JBoss Marshalling classloader fixes

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12824

### CODE REVIEW

File: jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java [hunk 1]
```diff
@@ -27,6 +27,12 @@ public GenericJBossMarshaller(ClassAllowList classAllowList) {
 
    public GenericJBossMarshaller(ClassLoader classLoader, ClassAllowList classAllowList) {
       super();
+      if (classLoader == null) {
+         classLoader = classAllowList != null ? classAllowList.getClassLoader() : null;
```
Comment (wburns) [inline jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java:31]: Shouldn't we prioritize the context class loader over the class loader of the class allow list?

**Review by wburns — COMMENTED** [review]
Changes in general look okay, but not sure of priority of which class loader to use.

File: jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java (hunk 1 above)
Comment (tristantarrant) [reply jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java:31]: Not sure :)

Comment (wburns) [reply jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java:31]: I worry that the ClassAllowList class will be from a different class loader than the user classes possibly.

Comment (tristantarrant) [reply jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java:31]: Ah, ClassAllowList's getClassLoader() is NOT ClassAllowList.class.getClassLoader() :)

Comment (wburns) [reply jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java:31]: Aha, my mistake. I was looking at my branch which doesn't have https://github.com/infinispan/infinispan/commit/ad05d66f4e061ea0e4b977b7b591e5a6f9e953b8 in it. It was integrated after I was done.

**Comment by wburns** [comment]
Integrated into master, thanks @tristantarrant !

**Pull request merged**